src/btree/bt_walk.c

```diff
@@ -40,21 +40,15 @@ retry:	WT_INTL_INDEX_GET(session, ref->home, pindex);
 	 * is slower.
 	 */
 	i = ref->pindex_hint;
-	if (i < pindex->entries && pindex->index[i] == ref) {
-		*pindexp = pindex;
-		*slotp = i;
-		return;
-	}
+	if (i < pindex->entries && pindex->index[i] == ref)
+		goto found;
 	while (++i < pindex->entries)
-		if (pindex->index[i] == ref) {
-			*pindexp = pindex;
-			*slotp = ref->pindex_hint = i;
-			return;
-		}
+		if (pindex->index[i] == ref)
+			goto found;
 	for (i = 0; i < pindex->entries; ++i)
 		if (pindex->index[i] == ref) {
-			*pindexp = pindex;
-			*slotp = ref->pindex_hint = i;
+found:			*pindexp = pindex;
```

Comment:
Did you deliberately leave out setting the hint altogether now? I'd sort of expected there to be a new line with:

```
ref->pindex_hint = *slotp;
```